### DISCUSSION TITLE
Custom Bicycle Route Planning

### DISCUSSION THREAD

**Discussion by sklv7** [organicmaps/organicmaps#8681] [Ideas]
When navigating by bike where I live, the app takes the most direct streets which usually involves high speed traffic. Taking side-streets and more secluded paths is how I prefer to travel, but currently I can’t plan turn-by-turn navigation with my own route. Adding stops between the _start_ and _destination_ can help guide the navigation, but it isn’t very customizable. 

•The feature I’m imagining is an ability to “draw” a custom path to the destination; allowing me to stick to quieter scenic roads and choose my exact route. 

I’m sure this feature could also be useful for hikers wanting to take specific paths. Being able to save the routes created on my device would be ideal and preserve privacy (no iCloud sync necessary).

**Comment by hb0nd** [comment]
https://github.com/organicmaps/organicmaps/issues/1360 is the navigation half the issue

Drawing paths in-app is an interesting idea, there are similar tools like https://www.gpsvisualizer.com/draw/

**Reply by biodranik** [reply]
Would it be better to automate and simplify the planning by using some common/popular algorithmic criteria to choose the route? How would you formulate "quieter scenic roads"?

**Reply by sklv7** [reply]
I don’t pretend to know how it would work, but there is a French bike app called Geovelo that has options for “fast” and “safe” routes. Unfortunately it doesn’t perform as well where I live now. 

For automatic routing options, I’m guessing that taking more indirect paths away from busy intersections would be a large part of the solution. 

•For example there’s a creek trail near me, but maps usually avoids it and directs me on the main streets with a direct route
